routingv8/request.go

```diff
@@ -0,0 +1,327 @@
+package routingv8
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type CalculateMatrixBody struct {
+	// Origins defining start points of the routes in the matrix.
+	// See https://developer.here.com/documentation/matrix-routing-api/8.6.0/dev_guide/topics/modes/modes.html
+	// for guidance on the matrix limitations.
+	Origins []*GeoWaypoint `json:"origins"`
+	// Destinations defining destinations of the routes in the matrix.
+	// See https://developer.here.com/documentation/matrix-routing-api/8.6.0/dev_guide/topics/modes/modes.html
+	// for guidance on the matrix limitations.
+	Destinations []*GeoWaypoint `json:"destinations"`
+	// Time of departure for all origins. Default to now.
+	DepartureTime string `json:"departureTime,omitempty"`
+	// Region of where the matrix should be calculated.
+	RegionDefinition RegionDefinition `json:"regionDefinition"`
+	// Profile to use for route calculation in the matrix.
+	Profile Profile `json:"profile,omitempty"`
+	// Route optimization.
+	RoutingMode RoutingMode `json:"routingMode,omitempty"`
+	// Vehicle to use.
+	TransportMode TransportMode `json:"transportMode,omitempty"`
+	// Attributes to receive back in the response.
```

Comment:
GoDoc comments should typically always begin with the thing they are describing, i.e. `// MatrixAttributes`